website/public/js/controllers/groupsCtrl.js

```diff
@@ -308,6 +308,11 @@ habitrpg.controller("GroupsCtrl", ['$scope', '$rootScope', 'Shared', 'Groups', '
       return message.highlight;
     }
 
+    $scope.togglehidingSystemMessages = function() {
+      // Intentionally flip the boolean state
+      $scope.hideSystemMessages = $scope.hideSystemMessages ? false : true;
```

Comment:
Agreed. My logic was that since `hideSystemMessages` is initially undefined, that I would want to explicitly make it a boolean. However, your suggestion is much cleaner:
```
> var test;
undefined
> test = !test;
true
> test = !test;
false
```